ydb/library/yql/minikql/comp_nodes/ut/mkql_block_map_join_ut.cpp

```diff
@@ -273,35 +273,50 @@ void RunTestBlockJoin(TSetup<false>& setup, EJoinKind joinKind,
 //
 // Auxiliary routines to build list nodes from the given vectors.
 //
+template <typename Type>
+struct TTypeMapper {
```

Comment:
Nicely done! Anyway, I suggest to enhance your approach the following way:
* Add the constructor with `TProgramBuilder&` parameter;
* Add the field to store the MKQL type, calculated in the introduced constructor;
* Convert the host (i.e. C++) value into the MKQL one via `GetValue` method (similar to `operator()`);
* Yield the calculated MKQL type via `GetType` method (similar to `*operator()`).

I guess, will make the code more readable and incapsulate all the if-else hell I wrote before. Thoughts?

Reply:
For the history: [fixed](https://github.com/ydb-platform/ydb/pull/9847/commits/4e9c7b00db240879cea096243deeba7e8b2bfb5a).